### DISCUSSION TITLE
[Bug/Compatibility issue] Obsidian wiki links break when syncing

### DISCUSSION THREAD

**Discussion by carbeer** [windingwind/zotero-better-notes#440] [Ideas]
**Describe the bug**
When adding an internal wiki link in a note (e.g. `[[link to some other note]]`, then the syncing process escapes the brackets with a backslash. As a result, it is not possible to use the link in Obsidian. 

**To Reproduce**
Add an internal note link in Obsidian using `[[]]` and invoke the syncing process.

**Expected behavior**
I see two options:

1. Do not escape the brackets when syncing
2. Convert the links into the HTML format that is compatible with the plugin. I think this should not be terribly complicated, as there already is some similar code that converts Obsidian images with a `![[]]` syntax to HTML images.

**Comment by windingwind** [comment]
I'm sorry for not getting back to you sooner. I don't know what the use case of keeping brackets is, as they may spoil the syncing. MD style link/image in the note will be converted to an actual MD link/image in the exported MD file, and when it is synced back, it becomes an actual link/image.

See #301

**Comment by carbeer** [comment]
Hi, sorry for my delayed response.
Wiki-style links `[[like this]]`  are the default way to link to other documents within Obsidian. As of now, during the MD to Zotero syncing process, they are converted to something `\[\[like this\]\]` (if I remember correctly). Therefore, the link can be used neither in Obsidian nor in Better Notes.

If these wiki-style links could be converted into proper markdown links `[like this](some file path/like this.md)` during the syncing process, this would greatly improve the current situation. 
Parsing it (optionally, depending on settings) back to wiki-stlye links during Zotero to MD sync would be amazing, but that's definitely not a priority, as regular markdown links can also be represented in Obsidian. 

I think similar functionality is already implemented for `![[images]]`. I have played around with the code a bit and have seen that these are converted into markdown links with a correct path. So adding this functionality would mostly involve another parsing step, using existing functionality. I have played around with the code a bit, but didn't get it to work properly yet and my changes felt quite hacky, as I didn't fully understand the parsing workflow.  

I hope this clarifies my request a bit?

**Comment by windingwind** [comment]
I see. The conversion of [[]] links will be partly supported in the future.

Only non-zotero links can be converted to [[]].

**Comment by windingwind** [comment]
Keeping brackets is another thing, which I hope we should never touch. It makes syncing uncontrollable, as I states above.

Thanks for your detailed reply and tries with the code. Recently there will be a total refactoring, which should make the code more readable and clearer.

**Comment by carbeer** [comment]
That sounds great, thanks a lot for your efforts!

**Comment by windingwind** [comment]
Please see #429. Welcome contribution on that branch for compatibility of Ob wiki links.
